dbgd.c

```diff
@@ -211,21 +211,55 @@ static int dbgd_mem_read(Dbg__Request *req, Dbg__Response *res)
     res->address = req->address;
     res->has_address = 1;
 
-    res->size = 1; /* FIXME: add word, dword, qword support */
-    res->has_size = 1;
+    res->data.len  = req->size;
+    res->data.data = malloc(res->data.len);
```

Comment:
`free`?

Reply:
See "Known problems" in first post. [We discussed this in 2017 when I prototyped this PR but I never got a response](https://github.com/JayFoxRox/nxdk-rdt/pull/7#discussion_r126584315).

I know this is bad code, but without this feature (MMIO compatible read/write) nxdk-rdt is pretty useless. I don't really want to spend another few hours to rewrite this differently - this code already exists and at least makes nxdk-rdt functional.
We can iterate and make nxdk-rdt less leaky and more stable later.